### ISSUE TITLE
Why `extended_attention_mask` should broadcast to `[bs, 1, seqlen, seqlen]`?

### ISSUE THREAD

**Issue by MenglingD** [NVIDIA/FasterTransformer#260]
The`extended_attention_mask` of `HuggingFaceEncoder` in `examples/pytorch/bert_example.py` is diff from the encoder of bert from huggingface。
In `examples/pytorch/bert_example.py`，the `extended_attention_mask` is not only broadcasted at dim3 and at dim2 but also mask at dim2，for example:
```python3
batch_size = 1
seq_len = 2
attention_mask = [[1, 0]]
extended_attention_mask = [[[[1, 0], [0, 0]]]]  # shape：[batch_size, 1, seqlen, seqlen]，1 will broadcast with head_num, and the first seqlen dim is masked
```
But in bert from huggingface，the extended_attention_mask is only broadcasted at dim3：
```python3
attention_mask = [[1, 0]]
extended_attention_mask = [[[[1, 0]]]]  # shape：[batch_size, 1, 1, seqlen]，1 will broadcast with head_size
# broadcast to seqlen at dim2 as add to attention_score
extended_attention_mask = [[[[1, 0], [1, 0]]]  # dim2 is only broadcasted
```
the codes in huggingface is [modeling_bert.py#L348](https://github.com/huggingface/transformers/blob/main/src/transformers/models/bert/modeling_bert.py#L348).

The values of `extended_attention_mask` in `examples/pytorch/bert_example.py` is diff from of bert from huggingface：`[[[[1, 0], [0, 0]]]] vs [[[[1, 0], [1, 0]]]`，I am curious about that why the `extended_attention_mask` value is difference which will leads hidden_states dismatch.

**Comment by byshiue** [comment]
Because in EFF of FT (remove padding), we only compute the sentences of [[1, 0], [0, 0]]. Namely, first token of first sentence. If the mask is [[1, 0], [1, 0]] the results of padding parts would be different. 
If we don't care the padding part, then both [[1, 0], [1, 0]] and [[1, 0], [0, 0]] should work and generate same results except the padding part.
